lib/attr_comparable.rb

```diff
@@ -34,6 +36,8 @@ def attr_compare(*attributes)
 
       class_eval <<-EOS
         def <=>(rhs)
+          return nil if rhs.nil?
```

Comment:
I don't think you need `return` here, so it's always best in Ruby to omit it:
```
unless rhs.nil?
  #{ attr_exprs.join(".nonzero? || ") }
end
```
(The value of `if` and `unless` is nil when they don't run their code block.)